### PR TITLE
P2448R2 Relaxing some constexpr restrictions

### LINKED ISSUE #5574: [2022-07 CWG Motion 9] P2448R2 Relaxing some constexpr restrictions
[P2448R2](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2448r2.html)

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #5574
Fixes cplusplus/papers#1119

### CODE REVIEW

File: source/declarations.tex [hunk 1]
```diff
@@ -816,78 +810,6 @@
...
-
-struct D : B {
-  constexpr D() : B(global) { }         // ill-formed, no diagnostic required
-                                        // lvalue-to-rvalue conversion on non-constant \tcode{global}
-};
-
-constexpr int f(int x) {
-  static int n = x;
-  return n + x;                         // ill-formed, no diagnostic required
-                                        // all calls reach the static variable declaration
```
Comment (tkoeppe) [inline source/declarations.tex:886]: This change isn't in the paper (which I suppose has a stale baseline). This example was added by https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2242r3.html.

@jensmaurer Is it OK to lose this example?

Comment (jensmaurer) [reply source/declarations.tex:886]: I think we want to keep that example. Can you revert that hunk?

Comment (tkoeppe) [reply source/declarations.tex:886]: We've lost the ambient paragraph, though, so where should the residual example be? Just in a paragraph of its own?

Comment (JohelEGP) [reply source/declarations.tex:886]: But the paper normatively removes the restriction that it's IFNDR, so the example no longer has normative backing. If it does have, the example can be moved to where that is justified.

Comment (tkoeppe) [reply source/declarations.tex:886]: I see, thanks. @jensmaurer OK to lose this then?

Comment (jensmaurer) [reply source/declarations.tex:886]: Right, ok with keeping the example removed.

Comment (tkoeppe) [reply source/declarations.tex:886]: Great, thank you both!

**Pull request merged**